Carthage Support (cont.)

toohotz opened the pull request:
This PR is a continuation of PR # 270 from @NicholasTD07 adding Carthage support.

**Resolved issues:**

* Changed the two previous public gesture methods to private.

* Also resolved a method deprecation issue.

toohotz commented:
For future reference, it might be best to actually go with **public**  on the gesture methods as in the the long run as in Xcode 7 beta 6, this no longer works in Swift 2.0 as the compiler now complains about the Obj-C protocol being used here.  

Only noticed this as I was spending time converting the swift files from 1.2 to Swift 2.0.

![screen shot 2015-08-27 at 1 40 27 pm](https://cloud.githubusercontent.com/assets/795804/9528208/6bc897e4-4cc1-11e5-9cfa-0cbc05948066.png)

hackiftekhar commented:
Hi @toohotz, I already tried doing gesture methods as private, and It seems to be crashed 'unrecognized selector' when using with 'shiuldResignTouchOutside' to true configuration. FYI I'm ok with public too if it causing crash.

One more thing about interfaceOrientation, is there any reason except deprecated warning for asking about interfaceOrientation from statusBar. It may cause library logic failure, We must need to test it with all scenarios before replacing interfaceOrientation to statusBar orientation. Possibly we can discuss, test and fix it separately on #229.

toohotz commented:
@hackiftekhar good point, I've reverted the change for now and addressed the public declaration issue.  We can test the different scenarios in that thread and make a PR once ready for it.
